### DISCUSSION TITLE
Support for multiple GPUs

### DISCUSSION THREAD

**Discussion by Kevin3263** [FLAMEGPU/FLAMEGPU2#1065] [General]
Hi,

We've been implementing a covid-19 epidemic simulation model with FLAME. The model runs extraordinarily fast but the scope is limited by the size of VRAM on a single GPU. In our model, each agent takes up around 200 bytes of space, so the maximum number of agents supported in the model is limited to 400 million on a 80GB H100, which has the biggest VRAM currently. Even though the model will still run super fast(less than a second per step), we cannot further expand its scope.

We would like to ask if FLAME supports multiple GPUs of plans to do that in the future?

Many thanks,
Kevin

**Comment by mondus** [comment]
Hi Kevin,

Its an interesting question. It is not an easy task as it requires careful consideration of how to undertake partitioning and load balancing. We have plans on how this could be done (mostly utilizing ideas form FLAME GPU's predecessor FLAME which ran on distributed CPUs). We previously applied for research funding to develop the functionality to add multiple GPU support but missed out by a fine margin. The most realistic way that we will get a sizable chunk of developer time to do this is via collaborative research funding. This is how the majority of FLAME GPU has been developed over a number of years. If you are part of a research group we could perhaps discuss how this could work but I suggest to move the conversation to email. Feel free to drop me an email as I would love to find out a bit more about how you are using FLAME GPU. 400 million agents is rather impressive!

Paul

**Comment by mondus** [comment]
In terms of immediate ways in which you could scale your model further. We could look to supporting managed memory which would allow the whole of host memory to be utilized (with performance implications). Alternatively there may be ways in which the size of your agents could be reduced. This would require us to understand more about your model. Do you have anything published?
